In [4]:
import numpy as np
import pandas as pd
acc_intake = pd.read_csv('aac_intakes.csv', index_col=1)


In [5]:
#acc_intake.dtypes

### 1) Find the top 5 places where animals are found so the shelter can coordinate with local volunteers and animal control to monitor these areas. 

In [6]:
#location = acc_intake.groupby("found_location")["animal_id"].count()
#location.sort_values(ascending=False).head(20)
acc_intake.found_location.value_counts().head(5)

Austin (TX)                          14443
Outside Jurisdiction                   948
Travis (TX)                            921
7201 Levander Loop in Austin (TX)      517
Del Valle (TX)                         411
Name: found_location, dtype: int64

#### Answer 1 ) Above it is shown the top 5 places where animals are found. 
To reach these results, I used value_counts(), that returns a series containing counts of unique values (in this case, intake_found_location); and the function head(), that returns the first n rows (in this case, 5 rows) for the object based on position.

### 2) What is the average number of pets found in a month in the year 2015? Are there months where there is a higher number of animals found?

In [9]:
acc_intake_outcomes = pd.read_csv('aac_intakes_outcomes.csv',index_col=1)

In [124]:
year2015 = acc_intake_outcomes.loc[(acc_intake_outcomes['intake_year'] == 2015)].count()
permonth = acc_intake_outcomes.groupby("intake_month")["intake_year"].count()
print("List of months in descending order")
print(permonth.sort_values(ascending=False))
print()
print("Average number of pets found in a month in the year 2015 =", permonth.mean())


List of months in descending order
intake_month
5     7966
10    7652
6     7459
3     6613
11    6597
7     6528
8     6498
9     6489
4     6112
12    6099
1     5982
2     5677
Name: intake_year, dtype: int64

Average number of pets found in a month in the year 2015 = 6639.333333333333


#### Answer 2 ) Above it is listed the count for each month for 2015. The average across those months is shown below: around 6639 ( corroborated by the fact that most months are all in the 5/6/7 thousands)
To reach these results, I first created a variable year2015 to store just values where intake_year was equal to 2015. Then I grouped these values (using group_by)  by month (intake_month) and counted unique values (function count()) inside each group (here, group of months).
Using mean() I got the avearge of the values over the requested axis.

### 3) What is the ratio of incoming pets vs. adopted pets?
#### This key metric helps the shelter know how they are doing.


In [107]:
adopted = acc_intake_outcomes.loc[(acc_intake_outcomes['outcome_type'] == "Adoption")].count()
total = acc_intake_outcomes["count"].count()
print("Total number of income pets - ", total)
print("Total number of adopted pets -", adopted["count"])
print("Ratio between total and adopted - ", adopted["count"]/total)

Total number of income pets -  79672
Total number of adopted pets - 33594
Ratio between total and adopted -  0.4216537805000502


#### Answer 3 ) Ratio is the comparison between two numbers  (usually by dividing them) and it describes roughly how many times one value can contain another.  Hence,  the ratio between incoming pets and adopted pets would be the total number of incoming pets divided by the total number of adopted pets. 

For clarification, I also printed the total number of incoming pets and adopted pets. A ratio of 0.42 shows that less than half of all the incoming pets are adopted.

### 4) What is the distribution of the types of animals in the shelter?
#### Find the count of each type of animal in the shelter.


In [481]:
types = acc_intake_outcomes.animal_type.value_counts()
print(types)


Dog      45366
Cat      29539
Other     4428
Bird       339
Name: animal_type, dtype: int64


####  Answer 4 ) Above it is the count of each type of animal in the shelter. 

To reach these results, I created a variable called "types" to store all types of animals; and used value_counts(), that returns a series containing counts of unique values (in this case, animal_type).

### 5) What are the adoption rates for specific breeds?
#### Find the top 5 dog breeds in the shelter (based on count) and then find the adoption percentage of each breed.


In [96]:
breed = acc_intake_outcomes.breed.value_counts().head(5)
print("TOP 5 DOG BREED IN THE SHELTER")
print()
print(breed)
adopted = acc_intake_outcomes.loc[(acc_intake_outcomes['outcome_type'] == "Adoption")]
shorthair = adopted.loc[(adopted['breed'] == "Domestic Shorthair Mix")]
pitbull = adopted.loc[(adopted['breed'] == "Pit Bull Mix")]
chihuahua = adopted.loc[(adopted['breed'] == "Chihuahua Shorthair Mix")]
labrador = adopted.loc[(adopted['breed'] == "Labrador Retriever Mix")]
mediumhair = adopted.loc[(adopted['breed'] == "Domestic Medium Hair Mix")]
print()
print("ADOPTION PERCENTAGE OF EACH BREED")
print()
print("There were", shorthair["count"].count(), "Domestic Shorthair Mix adopted from a total of 8270: a percentage of", (shorthair["count"].count()/23423)*100 )
print("There were" , pitbull["count"].count(), "Pit Bull Mix adopted from a total of 6673: a percentage of", (pitbull["count"].count()/6256)*100)
print("There were", chihuahua["count"].count(),"Chihuahua Shorthair Mix adopted from a total of 4471: a percentage of", (chihuahua["count"].count()/4831)*100)
print("There were",  labrador["count"].count(), "Labrador Retriever Mix adopted from a total of 3598: a percentage of", (labrador["count"].count()/4789)*100)
print("There were", mediumhair["count"].count(), "Domestic Medium Hair Mix adopted from a total of 2835: a percentage of", (mediumhair["count"].count()/2326)*100)


TOP 5 DOG BREED IN THE SHELTER

Domestic Shorthair Mix      23423
Pit Bull Mix                 6256
Chihuahua Shorthair Mix      4831
Labrador Retriever Mix       4789
Domestic Medium Hair Mix     2326
Name: breed, dtype: int64

ADOPTION PERCENTAGE OF EACH BREED

There were 9865 Domestic Shorthair Mix adopted from a total of 8270: a percentage of 42.11672287922128
There were 2324 Pit Bull Mix adopted from a total of 6673: a percentage of 37.148337595907925
There were 2270 Chihuahua Shorthair Mix adopted from a total of 4471: a percentage of 46.98820120057959
There were 2368 Labrador Retriever Mix adopted from a total of 3598: a percentage of 49.446648569638754
There were 1037 Domestic Medium Hair Mix adopted from a total of 2835: a percentage of 44.58297506448839


####  Answer 5 ) Above are listed the adoption rates for the top 5 breeds in the shelter

To reach these results, I created a variable called "breed" to store the top 5 breeds in the shelter and used value_counts(), that returns a series containing counts of unique values (in this case, breed); and the function head(), that returns the first n rows (in this case, 5 rows) for the object based on position. Hence, the variable "breed" shows the total number of individuals in the top 5 breed category.

Then I created another variable called adopted, where I stored just values where outcome_type was equal to Adoption. Hence, the variable "adopted" stores only adopted animals.

Within the adopted variable, I search for the amount of adopted animals from the top 5 breeds. To get the ratio, I divided the number of adopted by the total number of individuals and multipled by 100 to get the percentage.

### 6) What are the adoption rates for different colorings?
#### Find the top 5 colorings in the shelter (based on count) and then find the adoption percentage of each color.

In [104]:
color = acc_intake_outcomes.color.value_counts().head(5)
print("TOP 5 COLORINGS IN THE SHELTER")
print()
print(color)
adopted = acc_intake_outcomes.loc[(acc_intake_outcomes['outcome_type'] == "Adoption")]
blackwhite = adopted.loc[(adopted['color'] == "Black/White")]
black = adopted.loc[(adopted['color'] == "Black")]
browntabby = adopted.loc[(adopted['color'] == "Brown Tabby")]
brown = adopted.loc[(adopted['color'] == "Brown")]
white = adopted.loc[(adopted['color'] == "White")]
print()
print("ADOPTION RATIO OF EACH COLOR")
print()
print("There were", blackwhite["count"].count(), "Black/White adopted from a total of 8270: a percentage of", (blackwhite["count"].count()/8270)*100)
print("There were" , black["count"].count(), "Black adopted from a total of 6673: a percentage of", (black["count"].count()/8270)*100)
print("There were", browntabby["count"].count(),"Brown Tabby adopted from a total of 4471: a percentage of", (browntabby["count"].count()/8270)*100)
print("There were",  brown["count"].count(), "Brown adopted from a total of 3598: a percentage of", (brown["count"].count()/8270)*100)
print("There were", white["count"].count(), "White Tabby adopted from a total of 2835: a percentage of", (white["count"].count()/8270)*100)




TOP 5 COLORINGS IN THE SHELTER

Black/White    8270
Black          6673
Brown Tabby    4471
Brown          3598
White          2835
Name: color, dtype: int64

ADOPTION RATIO OF EACH COLOR

There were 3738 Black/White adopted from a total of 8270: a percentage of 45.19951632406288
There were 2692 Black adopted from a total of 6673: a percentage of 32.55139056831923
There were 1871 Brown Tabby adopted from a total of 4471: a percentage of 22.623941958887546
There were 794 Brown adopted from a total of 3598: a percentage of 9.600967351874244
There were 1065 White Tabby adopted from a total of 2835: a percentage of 12.877871825876664


####  Answer 6 ) Above are listed the adoption percentages for the top 5 colorings in the shelter

To reach these results, I created a variable called "color" to store the top 5 colorings in the shelter and used value_counts(), that returns a series containing counts of unique values (in this case, color); and the function head(), that returns the first n rows (in this case, 5 rows) for the object based on position. Hence, the variable "color" shows the total number of individuals in the top 5 color category.

Then I created another variable called adopted, where I stored just values where outcome_type was equal to Adoption. Hence, the variable "adopted" stores only adopted animals.

Within the adopted variable, I search for the amount of adopted animals from the top 5 colorings.  To get the ratio, I divided the number of adopted by the total number of individuals and multipled by 100 to get the percentage.


### 7) About how many animals are spayed/neutered each month?
#### This will help the shelter allocate resources and staff. Assume that all intact males and females will be spayed/neutered.


In [35]:
sex = acc_intake_outcomes.loc[(acc_intake_outcomes['sex_upon_intake'] == "Intact Male") | (acc_intake_outcomes['sex_upon_intake'] == "Intact Female" )] 
print(sex['sex_upon_intake'].count())
months = sex.groupby("intake_month").count()
print(months["count"])

49021
intake_month
1     3521
2     3184
3     3399
4     3531
5     5211
6     4994
7     4287
8     4121
9     4184
10    4860
11    4087
12    3642
Name: count, dtype: int64


##### 7 Answer ) Above it is shown many animals are spayed/neutered each month  - first number (49021) is the count of pets that are intact on intake, and below it is the  rundown of how many there are per month. 

To reach these results, I first created a variable sex to store just values where sex_upon_intake was equal to "Intact Male" OR  "Intact Female". Then I grouped these values (using group_by) by month (intake_month) and counted unique values (function count()) inside each group (here, group of months).